src/Decompiler/Analysis/SsaTransform.cs

```diff
@@ -490,18 +490,32 @@ private void GenerateUseDefsForKnownCallee(CallInstruction ci, Procedure callee,
                 }
                 //$REVIEW: this is very x86/x87 specific; find a way to generalize
                 // this to any sort of stack-based discipline.
+                var fpuDefs = new Dictionary<FpuStackStorage, Identifier>();
                 foreach (var def in calleeFlow.Trashed.OfType<FpuStackStorage>()
                     .Where(def => def.FpuStackOffset >= fpuStackDelta))
                 {
-                    var fpuDefExpr = arch.CreateFpuStackAccess(
-                        ssa.Procedure.Frame, 
-                        def.FpuStackOffset,
-                        PrimitiveType.Word64); //$TODO: datatype?
-                    fpuDefExpr = fpuDefExpr.Accept(this);
+                    var name = $"rRet{def.FpuStackOffset - fpuStackDelta}";
+                    var id = ssa.Procedure.Frame.CreateTemporary(
+                        name, PrimitiveType.Word64); //$TODO: datatype?
+                    var fpuDefId = NewDef(id, ci.Callee, false);
                     ci.Definitions.Add(
                         new CallBinding(
                             def,
-                            fpuDefExpr));
+                            fpuDefId));
+                    fpuDefs.Add(def, fpuDefId);
+                }
+                foreach(var fpuDef in fpuDefs)
+                {
+                    var fpuAccess = arch.CreateFpuStackAccess(
+                        ssa.Procedure.Frame,
+                        fpuDef.Key.FpuStackOffset,
+                        PrimitiveType.Word64); //$TODO: datatype?
+                    var iCur = stmCur.Block.Statements.IndexOf(stmCur);
+                    stmCur = stmCur.Block.Statements.Insert(
+                        iCur + 1,
+                        stmCur.LinearAddress,
+                        new Store(fpuAccess, fpuDef.Value));
+                    stmCur.Instruction = stmCur.Instruction.Accept(this);
                 }
```

Comment:
...refactor lines 504-518 to a new method `InsertFpuStackAccessAssignments`

You already have the unit tests in place, so refactoring should be safe.